In [1]:
import numpy as np
import cv2
from matplotlib import pyplot as plt

In [3]:
#the 1 at the end is telling opencv to download the image as a full color image
img = cv2.imread('house.jpg', 1)

If we pass in 0, we're trying to read the image as black and white.

In [4]:
img

array([[[ 99, 166, 175],
        [ 84, 152, 157],
        [ 83, 153, 153],
        ...,
        [221, 216, 207],
        [127, 122, 113],
        [ 91,  86,  77]],

       [[105, 169, 180],
        [ 86, 151, 159],
        [ 55, 125, 125],
        ...,
        [175, 169, 162],
        [131, 125, 118],
        [163, 157, 150]],

       [[113, 170, 185],
        [103, 165, 175],
        [ 52, 123, 127],
        ...,
        [105,  96,  93],
        [124, 115, 112],
        [194, 185, 182]],

       ...,

       [[ 55, 124,  97],
        [ 61, 129, 104],
        [ 52, 120,  97],
        ...,
        [  0,  94,  60],
        [  5,  95,  65],
        [ 26, 114,  84]],

       [[ 63, 134, 107],
        [ 44, 115,  88],
        [ 56, 121,  99],
        ...,
        [  3, 100,  68],
        [  0,  94,  63],
        [  7,  99,  70]],

       [[ 45, 119,  89],
        [ 56, 127, 100],
        [ 63, 128, 106],
        ...,
        [  7, 100,  71],
        [  9, 101,  72],
        [  3,  94,  68]]

The above cell displays all the pixels found in the image itself.

In [5]:
#displaying the datatype 
type(img)

numpy.ndarray

In [6]:
#total number of elements in the image
len(img)

863

We can see that the image is a list like object with 863 elements.

In [7]:
#diaplying the number of vertical columns
len(img[0])

1253

There are 1253 vertical columns.

In [8]:
#displaying the number of channels
len(img[0][0])

3

The above cell indicates there's an R, G, B channel.

If there was a transparency layer or alpha layer, it would be 4.

In [9]:
#rows, columns and channels
img.shape

(863, 1253, 3)

In [10]:
#type of image
img.dtype

dtype('uint8')

The image has an unsigned integer value of 8 which tells us it has a maximum of 2^8 values in each pixel.

The range of values vary from 0-255.

In [11]:
#displaying the pixel values
img[10,5]

array([ 84, 152, 151], dtype=uint8)

In [12]:
#total number of datapoints
img.size

3244017